### PR TITLE
Add some more convenience methods + make model forward calls type-safe

### PR DESCRIPTION
**Pull request by joeyballentine**

### CODE REVIEW

File: src/spandrel/__helpers/model_descriptor.py [hunk 1]
```diff
@@ -124,17 +124,52 @@ def __init__(
...
+    def cpu(self):
+        self.model.cpu()
+        return self
+
+    def eval(self):
+        self.model.eval()
+        return self
+
+    def __call__(self, *args, **kwargs):  # noqa: ANN002
+        return self.model(*args, **kwargs)
```
Comment (RunDevelopment) [inline src/spandrel/__helpers/model_descriptor.py:157]: No. This throws away any semblance of type safety.

File: src/spandrel/__helpers/model_descriptor.py [hunk 2]
```diff
@@ -124,17 +124,52 @@ def __init__(
...
+        """
+        Upscale the given image by the scale factor of the model.
+        """
+        return self.model(image)
 
 
 class FaceSRModelDescriptor(ModelBase[T], Generic[T]):
-    pass
+    def __call__(
+        self, image: torch.Tensor, return_rgb: bool = False, weight: float = 0.5
```
Comment (RunDevelopment) [inline src/spandrel/__helpers/model_descriptor.py:170]: What is `return_rgb` and `weight`? 

`outputChannels` decides the number of output channel, so what is `return_rgb`?

File: src/spandrel/__helpers/model_descriptor.py [hunk 3]
```diff
@@ -124,17 +124,52 @@ def __init__(
 
         self.model.load_state_dict(state_dict)  # type: ignore
 
-    def to(self, device: torch.device):
-        self.model.to(device)
+    def to(
+        self,
+        device: int | torch.device | None = torch.device("cpu"),
```
[the remark this answers is not in the document]
Comment (joeyballentine) [reply src/spandrel/__helpers/model_descriptor.py:129]: ... I am not a smart man

File: src/spandrel/__helpers/model_descriptor.py [hunk 4]
```diff
@@ -124,17 +124,52 @@ def __init__(
...
+        dtype: torch.dtype | str | None = None,
+        non_blocking: bool = False,
```
[the remark this answers is not in the document]
Comment (joeyballentine) [reply src/spandrel/__helpers/model_descriptor.py:131]: It's not used by us, but it mimics pytorch's API which is the reason I added it. Some day someone's gonna request this I bet

File: src/spandrel/__helpers/model_descriptor.py (hunk 1 above)
Comment (joeyballentine) [reply src/spandrel/__helpers/model_descriptor.py:157]: I'll make it throw an error then

File: src/spandrel/__helpers/model_descriptor.py (hunk 2 above)
Comment (joeyballentine) [reply src/spandrel/__helpers/model_descriptor.py:170]: It's used by the model code. https://github.com/chaiNNer-org/spandrel/blob/11aae676a9ee4dab7091424ef0ba7d9f7344f730/src/spandrel/architectures/GFPGAN/arch/gfpganv1_clean_arch.py#L302

Comment (joeyballentine) [reply src/spandrel/__helpers/model_descriptor.py:170]: Not really sure what return RGB is, but chaiNNer passes that into the method

File: src/spandrel/__helpers/model_descriptor.py (hunk 1 above)
Comment (RunDevelopment) [reply src/spandrel/__helpers/model_descriptor.py:157]: Why? Just remove it.

File: src/spandrel/__helpers/model_descriptor.py (hunk 2 above)
Comment (RunDevelopment) [reply src/spandrel/__helpers/model_descriptor.py:170]: Joey, an API is supposed to abstract over implementation details. The API for the call function you made is that it returns *one* image, so how is `return_rgb` relevant?

File: src/spandrel/__helpers/model_descriptor.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply src/spandrel/__helpers/model_descriptor.py:131]: Alright, I read up on `non_blocking`. Let's keep it.

**Comment by joeyballentine** [comment]
Closing this while we think of a better api design

**Pull request closed without merging**